lib/dependencies/AMDDefineDependencyParserPlugin.js

```diff
@@ -98,12 +105,14 @@ class AMDDefineDependencyParserPlugin {
 			return true;
 		}
 	}
-	processItem(parser, expr, param, namedModule) {
+	processItem(parser, expr, param, paramExpr, namedModule) {
 		if (param.isConditional()) {
-			param.options.forEach(param => {
-				const result = this.processItem(parser, expr, param);
+			param.options.forEach((param, idx) => {
```

Comment:
We need some tests for this.

I think this fails if multiple `?:` operators are nested. In this case `options` has more than 2 elements:

``` js
import(a ?
  (b ? "x" : `./${CONST_PREFIX}/${DEFINED_EXPRESSION}/${CONST_SUFFIX}`) :
  (c ? `./${CONST_PREFIX}/${DEFINED_EXPRESSION}/${CONST_SUFFIX}` : "y")
)
```

I think it make sense to add the expression tracking to the `BasicEvaluatedExpression` (i. e. `setExpression` for the whole thing, and `setTemplateString(quasis, parts)` for template strings), instead of reimplementing it here.

* You can call `setExpression` in `Parser.evaluateExpression` to do it once for all
* You need to change `getSimplifiedTemplateResult` in `Parser` a bit to get all the expressions too.
* `BasicEvaluatedExpression.setTemplateString` now takes `quasis` which is still the same, `parts` which are all the quasis and epxressions as `BasicEvaluatedExpression`s, so all the parts of the template string.

Reply:
Thanks! I've thought it's too heavy for this PR, but surely it's more appropriate for easily tracking back the expr from `BasicEvaluatedExpression`